File: client/src/app/shared/models/user-info-dto.ts
import { AddressDto } from "./address-dto";
import { UserRoles } from "./user-roles-enum";

export type UserInfoDto = 
{ 
    firstName: string;
    lastName: string;   
    email: string;
    address: AddressDto;
    role: UserRoles;
}

export interface LoginResponseDto extends UserInfoDto {
    
}

File: Core/DTOs/UserInfoDto.cs
using Core.Enums;
using System.ComponentModel.DataAnnotations;

namespace Core.DTOs
{
    public class UserInfoDto
    {
        public string? FirstName { get; set; }
        public string? LastName { get; set; }

        [EmailAddress]
        public string? Email { get; set; }
        public string? PhoneNumber { get; set; }
        public AddressDto? Address { get; set; }
        public UserRoles Role { get; set; }
    }
}
